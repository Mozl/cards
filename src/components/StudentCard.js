import React, { Component } from 'react';
import styled from 'styled-components';

const Box = styled.div`
  border: 0.1rem solid black;
  display: flex;
  flex-direction: column;
  text-color: black;
  border-radius: 1.25rem;
  padding: 1rem;
  margin-top: 1rem;
`;

const Wrapper = styled.div``;

const CardName = styled.div`
  display: flex;
  padding-left: 0.5rem;
  border: 0.1rem solid black;
  border-radius: 1.25rem;
  margin: 1rem 0 1rem 0;
  cursor: pointer;
  color: ${({ selected }) => (selected ? 'black' : 'white')};
  background-color: ${({ selected }) => (selected ? 'white' : 'grey')};
`;

CardName.displayName = 'StudentCardName';

class StudentCard extends Component {
  render() {
    const { selectStudentCard, selected, studentCredit } = this.props;
    return (
      <Wrapper>
        <CardName onClick={selectStudentCard} selected={selected}>
          <h2>Student Life</h2>
        </CardName>
        {!selected && (
          <Box>
            <p>Apr: 18.9%</p>
            <p>Balance Transfer Offer Duration: 0 months</p>
            <p>Purchase Offer Duration: 6 Months</p>
            <p>Credit Available: Â£{studentCredit}</p>
          </Box>
        )}
      </Wrapper>
    );
  }
}

export default StudentCard;
