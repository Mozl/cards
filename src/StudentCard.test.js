import React from 'react';
import ReactDOM from 'react-dom';
import StudentCard from './components/StudentCard';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<StudentCard />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<StudentCard />);
  expect(tree).toMatchSnapshot();
});

it('prop selected changes', () => {
  const wrapper = mount(<StudentCard selected={false} />);
  wrapper.setProps({ selected: true });
  expect(wrapper.props().selected).toEqual(true);
});
