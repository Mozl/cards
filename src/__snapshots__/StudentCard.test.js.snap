// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bwzfXH uTxCW"
>
  <div
    className="sc-htpNat dSltdS"
  >
    <h2>
      Student Life
    </h2>
  </div>
  <div
    className="sc-bdVaJa dIWyBO"
  >
    <p>
      Apr: 18.9%
    </p>
    <p>
      Balance Transfer Offer Duration: 0 months
    </p>
    <p>
      Purchase Offer Duration: 6 Months
    </p>
    <p>
      Credit Available: Â£
    </p>
  </div>
</div>
`;
